clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -300,12 +315,18 @@ pt2ir_t::convert(OUT instrlist_t **ilist)
             decode(GLOBAL_DCONTEXT, insn.raw, instr);
             instr_set_translation(instr, (app_pc)insn.ip);
             instr_allocate_raw_bits(GLOBAL_DCONTEXT, instr, insn.size);
+            instr_set_isa_mode(instr,
+                               insn.mode == ptem_32bit ? DR_ISA_IA32 : DR_ISA_AMD64);
             if (!instr_valid(instr)) {
-                ERRMSG("Failed to convert the libipt's IR to Dynamorio's IR.\n");
-                instrlist_clear_and_destroy(GLOBAL_DCONTEXT, *ilist);
-                return PT2IR_CONV_ERROR_DR_IR_CONVERT;
+                /* XXX:If the PT raw data contains some instructions not supported by
+                 * Dynamorio, we will not append them to the output instrlist_t.
+                 * Currently, the PT raw data may contains 'STAC' and 'CLAC' instructions.
+                 * We'd better to support them in the future.
```

Comment:
If we need to relax the failure strategy, I suggest adding the instr to the list. Users of pt2ir can decide what to do with invalid instrs. This also would prevent silent failures.

Also add a SYSLOG_INTERNAL_WARNING_ONCE to say that invalid instrs were found, and also a LOG to print the raw bytes of each bad instr.

Reply:
About "add a LOG to print the raw bytes of each bad instr", do you mean to add them in pt2ir.cpp or drpt2trace.cpp?

Reply:
Done.